app/code/community/Hps/Transit/Model/Source/AvsResultCodes.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * @category   Hps
+ * @package    Hps_Transit
+ * @copyright  Copyright (c) 2015 Heartland Payment Systems (https://www.magento.com)
+ * @license    https://github.com/hps/transit-magento-extension/blob/master/LICENSE  Custom License
+ */
+
+class Hps_Transit_Model_Source_AvsResultCodes
+{
+    public function toOptionArray()
+    {
+        return array(
+            array(
+                'value' => 'D2018',
+                'label' => 'AVS FAILED'
+            ),
+            array(
+                'value' => 'D2027',
+                'label' => 'AVS and CVV2 failed'
+            ),            
+        );
+    }
```

Comment:
We should be using the individual AVS result codes instead of the gateway's general response code.

Reply:
@slogsdon from the available transit documents which you already shared, I found only these 2 result codes related to AVS/CVV, is there any other document available for the result codes?

Reply:
You're performing the comparison against the `$response->avsResponseCode` and `$response->cvnResponseCode` response properties, which is correct, but the codes in these properties align with the standard AVS/CVV codes (i.e. what I listed in the requirements doc and what you're using in other PRs, [WooCommerce as an example](https://github.com/globalpayments/globalpayments-woocommerce/pull/4/files#diff-6763126723c42a6bca61c28e85e26cd28d4b0f20f14832e1e91c10f0b1a883a0R251-R282))

Reply:
ok.. updated the result codes